src/site/content/en/blog/bundling-non-js-resources/index.md

```diff
@@ -0,0 +1,193 @@
+
+It has similar tradeoffs to the [dynamic import](https://v8.dev/features/dynamic-import). While it's possible to use `import(...)` with arbitrary expressions like `import(someUrl)`, the bundlers give special treatment to a pattern with static URL `import('./some-static-url.js')` as a way to preprocess a dependency known at compile-time, yet [split it out into its own chunk](https://web.dev/reduce-javascript-payloads-with-code-splitting/) that's loaded dynamically.
+
+Similarly, you can use `new URL(...)` with arbitrary expressions like `new URL(relativeUrl, customAbsoluteBase)`, yet the `new URL('...', import.meta.url)` pattern is a clear signal for bundlers to preprocess and include a dependency alongside the main JavaScript.
+
+### Ambiguous relative URLs
+
+You might be wondering, why can't bundlers detect other common patterns—for example, `fetch('./module.wasm')` without the `new URL` wrappers?
+
+The reason is that, unlike import statements, any dynamic requests are resolved relatively to the document itself, and not to the current JavaScript file. Let's say you have the following structure:
+
+- index.html: `<script src="src/main.js" type="module"></script>`
+- src/
+  -  main.js
+  -  module.wasm
+
+If you want to load `module.wasm` from `main.js`, it might be tempting to use a relative path like `fetch('./module.wasm')`.
+
+However, `fetch` does not know the URL of the JavaScript file it's executed in, instead, it resolves URLs relatively to the document. As a result, `fetch('./module.wasm')` would end up trying to load `http://example.com/module.wasm` instead of the intended `http://example.com/src/module.wasm` and fail (or, worse, silently load a different resource than you intended).
+
+By wrapping the relative URL into `new URL('...', import.meta.url)` you can avoid this problem and guarantee that any provided URL is resolved relative to the URL of the current JavaScript module (`import.meta.url`) before it's passed on to any loaders.
+
+Replace `fetch('./module.wasm')` with `fetch(new URL('./module.wasm', import.meta.url))` and it will successfully load the expected WebAssembly module, as well as give bundlers a way to find those relative paths during the build time too.
+
+### Tooling support
+
+#### Bundlers
+
+Following bundlers support the `new URL` scheme already:
+
+-  [Webpack v5](https://webpack.js.org/guides/asset-modules/#url-assets)
+-  [Rollup](https://rollupjs.org/) (via plugins—[@web/rollup-plugin-import-meta-assets](https://modern-web.dev/docs/building/rollup-plugin-import-meta-assets/) for generic assets and [@surma/rollup-plugin-off-main-thread](https://github.com/surma/rollup-plugin-off-main-thread) for Workers specifically).
```

Comment:
```suggestion
-  [Rollup](https://rollupjs.org/) (Achieved via plugins—[@web/rollup-plugin-import-meta-assets](https://modern-web.dev/docs/building/rollup-plugin-import-meta-assets/) for generic assets and [@surma/rollup-plugin-off-main-thread](https://github.com/surma/rollup-plugin-off-main-thread) for Workers specifically.)
```

Reply:
Looking at this, which we discussed before, having it as a sentence looks neater, it's kind of an aside/clarification to the point.

Reply:
Thanks, yeah, this looks better.